remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java

```diff
@@ -116,13 +117,37 @@ public Collection<? extends SearchIndexedEntity> allIndexedEntities() {
       return mapping().allIndexedEntities();
    }
 
+   @Override
+   public Collection<? extends SearchIndexedEntity> indexedEntitiesForStatistics() {
+      long stamp = stampedLock.tryReadLock();
+      if (stamp == 0L) {
+         return Collections.emptySet();
+      }
+
+      try {
+         if (!searchMappingRef.available()) {
+            return Collections.emptySet();
+         }
+         return allIndexedEntities();
+      } finally {
+         stampedLock.unlockRead(stamp);
+      }
+   }
+
    @Override
    public Set<String> allIndexedEntityNames() {
       return mapping().allIndexedEntityNames();
    }
 
    @Override
    public Set<Class<?>> allIndexedEntityJavaClasses() {
+      if (!searchMappingRef.available()) {
```

Comment:
Looking at this class I an not sure what the stamped lock is protecting. It seems we can do the same `searchMappingRef.available()` call in indexedEntitiesForStatistics then?

Reply:
You right! Going to do that